includes/ingest.batch.inc

```diff
@@ -398,7 +398,9 @@ function islandora_batch_process_results($results, $timeout, $lock_timeout, &$co
     $ingest_object = unserialize($object['data']);
 
     if ($object['state'] !== ISLANDORA_BATCH_STATE__DONE) {
-      $object['state'] = $ingest_object->batchProcess();
+      // Both a simple state and state with message return are accepted.
+      $process_results = $ingest_object->batchProcess();
```

Comment:
Should probably update the abstract base, to indicate the new return type: https://github.com/Islandora/islandora_batch/blob/7.x/includes/islandora_batch_object_base.inc#L28-L37